server/cluster/manager.go

```diff
@@ -0,0 +1,265 @@
+			"clusterName:%s, schemaName:%s, tableName:%s, node:%s", clusterName, schemaName, tableName, node)
+	}
+
+	if _, err := cluster.CreateTable(ctx, schemaName, shardID, tableName, tableID); err != nil {
+		return 0, errors.Wrapf(err, "cluster manager AllocTableID, "+
+			"clusterName:%s, schemaName:%s, tableName:%s, node:%s", clusterName, schemaName, tableName, node)
+	}
+	return tableID, nil
+}
+
+func (m *ManagerImpl) GetTables(ctx context.Context, clusterName, node string, shardIDs []uint32) (map[uint32]*ShardTables, error) {
+	cluster, err := m.getCluster(ctx, clusterName)
+	if err != nil {
+		return nil, errors.Wrap(err, "cluster manager GetTables")
+	}
+
+	shardTablesWithRole, err := cluster.GetTables(ctx, shardIDs, node)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cluster manager GetTables, "+
+			"clusterName:%s, node:%s, shardIDs:%v", clusterName, node, shardIDs)
+	}
+
+	ret := make(map[uint32]*ShardTables, len(shardIDs))
+	for shardID, shardTables := range shardTablesWithRole {
+		tableInfos := make([]*TableInfo, len(shardTables.tables))
+
+		for _, t := range shardTables.tables {
+			tableInfos = append(tableInfos, &TableInfo{
+				id: t.meta.GetId(), name: t.meta.GetName(),
+				schemaID: t.schema.GetId(), schemaName: t.schema.GetName(),
+			})
+		}
+		ret[shardID] = &ShardTables{shardRole: shardTables.shardRole, tables: tableInfos, version: shardTables.version}
+	}
+	return ret, nil
+}
+
+func (m *ManagerImpl) DropTable(ctx context.Context, clusterName, schemaName, tableName string, tableID uint64) error {
+	cluster, err := m.getCluster(ctx, clusterName)
+	if err != nil {
+		return errors.Wrap(err, "cluster manager DropTable")
+	}
+
+	if err := cluster.DropTable(ctx, schemaName, tableName, tableID); err != nil {
+		return errors.Wrapf(err, "cluster manager DropTable, "+
+			"clusterName:%s, schemaName:%s, tableName:%s, tableID:%d", clusterName, schemaName, tableName, tableID)
+	}
+
+	return nil
+}
+
+func (m *ManagerImpl) RegisterNode(ctx context.Context, clusterName, node string, lease uint32) error {
+	nodePb := &clusterpb.Node{NodeStats: &clusterpb.NodeStats{Lease: lease}, Name: node}
+	cluster, err := m.getCluster(ctx, clusterName)
+	if err != nil {
+		return errors.Wrap(err, "cluster manager RegisterNode")
+	}
+	nodePb1, err := m.storage.CreateOrUpdateNode(ctx, cluster.clusterID, nodePb)
+	if err != nil {
+		return errors.Wrapf(err, "cluster manager RegisterNode, clusterName:%s, node:%s", clusterName, node)
+	}
+	cluster.Lock()
+	cluster.metaData.nodeMap[node] = nodePb1
+	cluster.Unlock()
+
+	// todo: refactor coordinator
+	if err := cluster.coordinator.Run(ctx); err != nil {
+		return errors.Wrap(err, "RegisterNode")
+	}
+	return nil
+}
+
+func (m *ManagerImpl) GetShards(ctx context.Context, clusterName, node string) ([]uint32, error) {
+	cluster, err := m.getCluster(ctx, clusterName)
+	if err != nil {
+		return nil, errors.Wrap(err, "cluster manager GetShards")
+	}
+	shardIDs, ok := cluster.nodes[node]
```

Comment:
Is this access goroutine-safe? Maybe we should avoid access to cluster inner data outside it.